IndexService independent from participant-state

gerolf-da opened the pull request:
The index service serves as a "bridge" between the data on the ledger and the ledger api users. That is also the reason why the index service API is defined in terms of the types in `ledger-api-domain`. It is the responsibility of the index service implementations to return compatible types and adhere to the expected semantics, so we can actually reuse index service implementations in the ledger api servers.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

gaboraranyossy-da commented on ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala:
```diff
@@ -0,0 +1,159 @@
+
+    final case class Create(
+        transactionId: TransactionId,
+        nodeId: NodeId,
```
do we actually need this? it looks like we don't need the eventId at all here: https://github.com/digital-asset/daml/pull/1314/files#diff-187da506c796f10dc0d43c95b946dd3fR37

gerolf-da replied:
Discussed with @dajmaki: Needs to be an opaque `String` that actually comes from the `ReadService`.

gerolf-da commented on ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/CompletionsService.scala:
```diff
@@ -0,0 +1,21 @@
+  **/
+trait CompletionsService {
+  def getCompletions(
+      beginAfter: Option[LedgerOffset],
```
should be `beginAt`.

gerolf-da commented on ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala:
```diff
@@ -0,0 +1,159 @@
+
+    final case class Archive(
+        transactionId: TransactionId,
+        nodeId: NodeId,
```
see https://github.com/digital-asset/daml/pull/1313/files#r286489644

gerolf-da commented on ledger/participant-state-index/src/main/scala/com/daml/ledger/participant/state/index/v2/package.scala:
```diff
@@ -0,0 +1,159 @@
+    */
+  final case class Transaction(
+      transactionData: GenTransaction.WithTxValue[NodeId, AbsoluteContractId],
+      blindingInfo: BlindingInfo,
```
Should not be needed. The filtering that needs this should happen within the index service.

gerolf-da commented:
* I changed the usage of `NodeId` to a new tagged type `AbsoluteNodeId` (better name suggestions are welcome, but we can change later as well).
* `bindlingInfo` is now removed from transaction. all filtering should happen within the transaction service.
